src/overlays/actors/ovl_En_Syateki_Niw/z_en_syateki_niw.c

```diff
@@ -103,296 +103,296 @@ void EnSyatekiNiw_Destroy(Actor* thisx, PlayState* play) {
                     }
 
-                    this->unk_2E8.x = this->unk_2DC.x + sp50;
-                    this->unk_2E8.z = this->unk_2DC.z + sp4C;
+                    this->targetPos.x = this->initPos.x + posXMod;
+                    this->targetPos.z = this->initPos.z + posZMod;
                     break;
             }
         } else {
-            this->unk_25C = 4;
+            this->hopTimer = 4;
             if (this->actor.bgCheckFlags & BGCHECKFLAG_GROUND) {
                 this->actor.velocity.y = 2.5f;
-                if ((Rand_ZeroFloat(10.0f) < 1.0f) && (this->unk_29E == 0)) {
-                    this->unk_25C = 0xC;
+                if ((Rand_ZeroFloat(10.0f) < 1.0f) && (this->minigameType == SYATEKI_MINIGAME_ARCHERY)) {
+                    this->hopTimer = 0xC;
                     this->actor.velocity.y = 10.0f;
                 }
             }
         }
     }
-    if (this->unk_25C != 0) {
+    if (this->hopTimer != 0) {
         sp4A = 1;
-        Math_ApproachF(&this->actor.world.pos.x, this->unk_2E8.x, 1.0f, this->unk_2C8.y);
-        Math_ApproachF(&this->actor.world.pos.z, this->unk_2E8.z, 1.0f, this->unk_2C8.y);
-        Math_ApproachF(&this->unk_2C8.y, 3.0f, 1.0f, 0.3f);
-        tmpf1 = this->unk_2E8.x - this->actor.world.pos.x;
-        tmpf2 = this->unk_2E8.z - this->actor.world.pos.z;
-
-        if (fabsf(tmpf1) < 10.0f) {
-            tmpf1 = 0;
+        Math_ApproachF(&this->actor.world.pos.x, this->targetPos.x, 1.0f, this->posRotStep.y);
+        Math_ApproachF(&this->actor.world.pos.z, this->targetPos.z, 1.0f, this->posRotStep.y);
+        Math_ApproachF(&this->posRotStep.y, 3.0f, 1.0f, 0.3f);
+        posXDiff = this->targetPos.x - this->actor.world.pos.x;
+        posZDiff = this->targetPos.z - this->actor.world.pos.z;
+
+        if (fabsf(posXDiff) < 10.0f) {
+            posXDiff = 0;
         }
 
-        if (fabsf(tmpf2) < 10.0f) {
-            tmpf2 = 0.0f;
+        if (fabsf(posZDiff) < 10.0f) {
+            posZDiff = 0.0f;
         }
 
-        if ((tmpf1 == 0.0f) && (tmpf2 == 0.0f)) {
-            this->unk_25C = 0;
-            this->unk_294 = 7;
+        if ((posXDiff == 0.0f) && (posZDiff == 0.0f)) {
+            this->hopTimer = 0;
+            this->targetPosTimer = 7;
         }
 
-        Math_SmoothStepToS(&this->actor.world.rot.y, RAD_TO_BINANG(Math_FAtan2F(tmpf1, tmpf2)), 3, this->unk_2C8.z, 0);
-        Math_ApproachF(&this->unk_2C8.z, 10000.0f, 1.0f, 1000.0f);
+        Math_SmoothStepToS(&this->actor.world.rot.y, RAD_TO_BINANG(Math_FAtan2F(posXDiff, posZDiff)), 3, this->posRotStep.z, 0);
+        Math_ApproachF(&this->posRotStep.z, 10000.0f, 1.0f, 1000.0f);
     }
 
-    if (this->unk_260 == 0) {
-        func_80B11A94(this, play, sp4A);
+    if (this->sootTimer == 0) {
+        EnSyatekiNiw_UpdateAnimationTarget(this, play, sp4A);
         return;
     }
 
-    if ((play->gameplayFrames % 4) == 0) {
+    if ((play->gameplayFrames % 4) == 0) { // draw smoke from bombhu hit
```

Comment:
with these docs I think it makes more sense to use an else instead of the early return
```c
} else if ((play->gameplayFrames % 4) == 0) { // draw smoke from bombchu hit
```